gnd/src/main/java/com/google/android/gnd/ui/home/HomeScreenViewModel.java

```diff
@@ -287,7 +282,18 @@ public void onFeatureClick(ImmutableList<MapFeature> mapFeatures) {
             .map(MapFeature::getFeature)
             .filter(f -> f != null)
             .collect(toImmutableList());
-    overlappingFeaturesSubject.onNext(features);
+
+    if (features.isEmpty()) {
+      Timber.e("onFeatureClick called with empty or null map features");
+      return;
+    }
+
+    if (features.size() == 1) {
```

Comment:
optional: Splitting hairs, but imo deciding when to show the bottom sheet is something that should be handled in the view not the view model, since we're making UI decisions. Maybe we can drop the bottom sheet behavior from the VM entirely, and restrict the VM to managing data (which is not too much in this case).

Reply:
In Jetpack I believe the recommendation is the keep as much logic as possible out of view, mostly because they're hard to unit test. In my understanding, views are really just for plumbing; the lowest UI layer close to Android SDK and OS. If this is a decision point the view needs to make, I believe it's correct to model in the in view model (thusly named).